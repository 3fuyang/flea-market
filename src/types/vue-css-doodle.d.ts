declare module '@luxdamore/vue-css-doodle'