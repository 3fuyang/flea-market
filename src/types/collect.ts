export class Collect {
  userID: string = ''
  goodID: string = ''
  collected: boolean = false
  collect () {
    if (!this.userID) {
      return '请先登录'
    } else if (this.userID.length === 4) {
      return '管理员请登录普通账号进行收藏'
    } else if (!this.goodID) {
      return '商品不存在'
    } else if (this.collected) {
      return '不能重复收藏'
    } else {
      return true
    }
  }
  set (uid: string = '', gid: string = '', collected: boolean = true) {
    this.userID = uid
    this.goodID = gid
    this.collected = collected
  }
}