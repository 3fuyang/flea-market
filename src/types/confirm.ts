export class Confirm {
  orderID: string = '1'
  stat: '待付款' | '待确认' | '待评价' | '已完成' = '待确认'
  confirm () {
    switch (this.stat) {
      case '待付款':
        return '请先付款'
      case '待确认':
        return true
      default:
        return '不可重复确认'
    }
  }
  set (st: '待付款' | '待确认' | '待评价' | '已完成' = '待确认') {
    this.stat = st
  }
}