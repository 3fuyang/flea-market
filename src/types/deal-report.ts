export class DealReport {
  adminID: string = '0001'
  orderID: string = '1'
  reported: '待处理' | '已封禁' | '已驳回' = '待处理'
  reply: string = ''
  dealReport (stat: '待处理' | '已封禁' | '已驳回') {
    if (this.reported !== '待处理') {
      return '不可重复处理举报'
    } else if (!this.reply) {
      return '请输入回复'
    } else {
      this.reported = stat
      return true
    }
  }
  set (rptd: '待处理' | '已封禁' | '已驳回', rpl: string) {
    this.reported = rptd
    this.reply = rpl
  }
}