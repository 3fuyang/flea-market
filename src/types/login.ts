// 测试用，类型建模
export class LoginForm {
  // 类型
  type: 'member' | 'admin' = 'member'
  // 用户名
  userID: string = ''
  // 密码
  password: string = ''
  // 登录
  login () {
    if (!this.userID) {
      return '请输入账号'
    } else if (this.type === 'member' && this.userID.length !== 7) {
      return '普通用户账号应长7位'
    } else if (this.type === 'admin' && this.userID.length !== 4) {
      return '管理员账号应长4位'
    } else if (!this.password) {
      return '请输入密码'
    } else {
      // 登录
      return true
    }
  }
  // setter
  set (tp: 'member' | 'admin' = 'member', id: string = '', pwd: string = '') {
    this.type = tp
    this.userID = id
    this.password = pwd
  }
}