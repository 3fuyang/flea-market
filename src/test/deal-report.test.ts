import { describe, it, expect } from "vitest"
import { DealReport } from "@/types/deal-report"

let counter = 1

const dealer = new DealReport()

const test = (rptd: '待处理' | '已封禁' | '已驳回', rpl: string, expc: string | boolean) => {
  return it(`cl_${counter++}`, () => {
    dealer.set(rptd, rpl)
    expect(dealer.dealReport('已封禁')).toBe(expc)
    expect(dealer).toMatchSnapshot()
  })
}

describe('report', () => {
  test('待处理', 'Something', true)
  test('待处理', '', '请输入回复')
  test('已驳回', 'Something', '不可重复处理举报')
  test('已驳回', '', '不可重复处理举报')
})