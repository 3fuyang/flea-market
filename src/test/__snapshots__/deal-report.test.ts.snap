// Vitest Snapshot v1

exports[`report > cl_1 1`] = `
DealReport {
  "adminID": "0001",
  "orderID": "1",
  "reply": "Something",
  "reported": "已封禁",
}
`;

exports[`report > cl_2 1`] = `
DealReport {
  "adminID": "0001",
  "orderID": "1",
  "reply": "",
  "reported": "待处理",
}
`;

exports[`report > cl_3 1`] = `
DealReport {
  "adminID": "0001",
  "orderID": "1",
  "reply": "Something",
  "reported": "已驳回",
}
`;

exports[`report > cl_4 1`] = `
DealReport {
  "adminID": "0001",
  "orderID": "1",
  "reply": "",
  "reported": "已驳回",
}
`;
